### PR TITLE
CI updates:

### PR DESCRIPTION
**Pull request by kshakir**
- Add dood to the ci docker compose used by jenkins.
- Run ci docker under user `hoggett`.
- Pass mysql creds to cromwell unit tests running in jenkins.
- Do not remove rendered creds via `sbt clean`.
- Reduce configs to `sbt test` and `sbt alltests:test`.
- Otherwise use the environment variable `CROMWEL_SBT_TEST_EXCLUDE_TAGS`.
- Tests don't need to be run in the directory `cromwell`.
- Only push artifacts on travis.
- Simplify skipping docker push.

### CODE REVIEW

**Comment by kshakir** [comment]
FYI:
- [dood](http://blog.teracy.com/2017/09/11/how-to-use-docker-in-docker-dind-and-docker-outside-of-docker-dood-for-local-ci-testing/)
- [hoggett](http://hero.wikia.com/wiki/Arthur_Hoggett)

**Comment by cjllanwarne** [comment]
Beyond tidiness, do you think this is going to have benefits for test reliability?

**Comment by kshakir** [comment]
Beyond tidiness, this PR for @aweng98 to run the unit tests in Jenkins. It won't affect reliability beyond allowing certain tests to pass. Here is a previous failed run without all of these changes: https://fc-jenkins.dsp-techops.broadinstitute.org/job/cromwell-test-runner/21/

File: project/Testing.scala [hunk 1]
```diff
@@ -4,62 +4,46 @@ import sbt.Keys._
...
-    // `nodocker:test` - Run all tests, except docker
-    testOptions in NoDockerTest := (testOptions in AllTests).value ++ Seq(DontUseDockerTaggedTests),
-    // `integration:test` - Run only integration tests
-    testOptions in CromwellIntegrationTest := (testOptions in AllTests).value ++
-      Seq(UseCromwellIntegrationTaggedTests, UseGcsIntegrationTaggedTests, UseAwsIntegrationTaggedTests),
-    // `nointegration:test` - Run all tests, except integration
-    testOptions in CromwellNoIntegrationTest := (testOptions in AllTests).value ++
-      Seq(DontUseCromwellIntegrationTaggedTests, DontUseGcsIntegrationTaggedTests, DontUseAwsIntegrationTaggedTests),
-    // `dbms:test` - Run database management tests.
-    testOptions in DbmsTest := (testOptions in AllTests).value ++ Seq(UseDbmsTaggedTests),
```
Comment (mcovarr) [inline project/Testing.scala]: sweet.

**Pull request merged**